lib/foreman_maintain/runner.rb

```diff
@@ -84,19 +89,19 @@ def storage
 
     private
 
-    def execute_scenario_steps(scenario)
+    def execute_scenario_steps(scenario, force = false)
       scenario.before_scenarios.flatten.each { |before_scenario| run_scenario(before_scenario) }
-      return if quit? # the before scenarios caused the stop of the execution
+      return if !force && quit? # the before scenarios caused the stop of the execution
```

Comment:
If we determine, if to fail or not based on the `fail_fast`/`fail_slow` options, I think we will achieve what you want.

Reply:
I found out that this force condition we have to keep as the `fail_fast`/`fail_slow` applies only on execution steps within scenario. Here we are between two scenarios.